innovation.game.php

```diff
@@ -129,6 +129,53 @@ function debug_draw($card_id) {
             throw new BgaUserException(self::format(self::_("This card is in {player_name}'s {location}"), array('player_name' => self::getPlayerNameFromId($card['owner']), 'location' => $card['location'])));
         }
     }
+    function debug_score($card_id) {
+        if (self::getGameStateValue('debug_mode') == 0) {
+            return; // Not in debug mode
+        }
+        $player_id = self::getCurrentPlayerId();
+        $card = self::getCardInfo($card_id);
+        $card['debug_score'] = true;
+        if ($card['location'] == 'hand' || $card['location'] == 'board' || $card['location'] == 'deck') {
+            self::transferCardFromTo($card, $player_id, 'score', false, true);
+        }
+        else if ($card['location'] == 'achievements') {
+            throw new BgaUserException(self::_("This card is used as an achievement"));
+        }
+        else if ($card['location'] == 'removed') {
+            throw new BgaUserException(self::_("This card is removed from the game"));
+        }
+        else {
+            throw new BgaUserException(self::format(self::_("This card is in {player_name}'s {location}"), array('player_name' => self::getPlayerNameFromId($card['owner']), 'location' => $card['location'])));
+        }
+    }
+    function debug_achieve($card_id) {
+        if (self::getGameStateValue('debug_mode') == 0) {
+            return; // Not in debug mode
+        }
+        $player_id = self::getCurrentPlayerId();
+        $card = self::getCardInfo($card_id);
+        $card['debug_achieve'] = true;
+        if ($card['location'] == 'hand' || $card['location'] == 'board' || $card['location'] == 'deck' || $card['location'] == 'score' || $card['location'] == 'achievements') {
+            try{
+                self::transferCardFromTo($card, $player_id, "achievements");
+            }
+            catch (EndOfGame $e) {
+                // End of the game: the exception has reached the highest level of code
+                self::trace('EOG bubbled from self::achieve');
```

Comment:
This should be:

self::trace('EOG bubbled from self::debug_achieve');

Reply:
Please update `self::trace('EOG bubbled from self::achieve');` to `self::trace('EOG bubbled from self::debug_achieve');`.